repoupdater.py

```diff
@@ -0,0 +1,134 @@
+import argparse
+import glob
+import os
+import re
+import sys
+import subprocess
+
+from github import Github
+from github.GithubException import GithubException
+import yaml
+
+import client
+
+ORG_NAME = "OpenSAFELY"
+REPO_NAME_PATTERN = "^.*-research$"
+BASE_PATH = os.path.abspath("research")
+
+
+def main():
+    parser = argparse.ArgumentParser(prog="repoupdater")
+    subparsers = parser.add_subparsers(help="sub-command help", dest="subcommand")
+    list_parser = subparsers.add_parser("list", help="list all repos")
+    update_parser = subparsers.add_parser(
+        "update", help="update all repos via pull or clone"
+    )
+    exec_parser = subparsers.add_parser(
+        "exec", help="execute command against all repos"
```

Comment:
see other comment (currently all subdirs, not all repos)

Reply:
subdirs ~= all checked out repos?

Reply:
yeah, the set of subdirs should contain the set of repos, but also contain some other stragglers, e.g. if you ran `$ python repoupdater.py exec git add .` but you have a `./venv/` lying around? not a big deal, though, will fail in an obvious way.